crates/egui/src/util/undoer.rs

```diff
@@ -89,6 +89,23 @@ impl<State> Undoer<State>
 where
     State: Clone + PartialEq,
 {
+    pub fn new() -> Self {
+        Self {
+            settings: Settings::default(),
+            undos: VecDeque::new(),
+            redos: Vec::new(),
+            flux: None,
+        }
+    }
```

Comment:
This is better done with an explicit `

```
impl<T> Default for Undoer<T> {
    …
}
```

It's only the derive macro (`#[derive(Default)]`) that fails when `T: !Default`

Reply:
Done